pkg/integrations/v2/ebpf/integration.go

```diff
@@ -0,0 +1,87 @@
+//go:build linux && ebpf_enabled
+// +build linux,ebpf_enabled
```

Comment:
IMO for now I'd ready have stuff like this be opt out (i.e., the default behavior is to build everything and you use tags to disable specific things). 

Can the tag be `!noebpf`? We use `nodocker` and `nonetwork` for disabling specific types of tests.

Reply:
The issue which prompted the opt-in tag is that the Linux agent builds would fail unless some specific system packages are installed (due to the Go->eBPF bindings dependency, which in turn requires [specific kernel versions](https://github.com/iovisor/bcc/blob/master/docs/kernel-versions.md)).

```
$ go build .          # on linux system without support for bcc
# github.com/iovisor/gobpf/bcc
../../vendor/github.com/iovisor/gobpf/bcc/module.go:32:10: fatal error: 'bcc/bcc_common.h' file not found
#include <bcc/bcc_common.h>
         ^~~~~~~~~~~~~~~~~~
1 error generated.
```

I'm not sure what's an elegant way to handle this. Bundling the required C headers, and somewhat complicating our build pipeline?

Reply:
You can update tools/seego/Dockerfile (it has apt, it's Debian-based) to install dependencies. We do this already in the main Dockerfile to install [systemd headers](https://github.com/rfratto/seego/blob/7257eaba401374ce0a6aa53a63317475aff3905c/Dockerfile#L85-L89), and extending the build image to include our specific changes would be the way to do this IMO. 

I think we can assume the kernel version is new enough, the platforms we'll be mainly building on should have kernels newer than ones from 2017.

Reply:
I see. We'll also need to add the dependencies when [linting](https://github.com/grafana/agent/blob/b7dd9dbd22f4eeaf3b00112651d5e52599f90280/.drone/drone.yml#L21). 

My main concern here is that people who build from source on Linux, might be confused when `go build` fails with that cryptic message. On one hand, they would already need the systemd headers as you mentioned, but I'm not sure that bcc is also something they'd expect to need if they're not using eBPF.

I'll try to find a way to build this and report back :)

Reply:
That's fair :) We can document it [in our contribution guide](https://github.com/grafana/agent/blob/main/docs/developer/contributing.md#compiling-the-agent) (we should probably be mentioning systemd there too but we're not at the moment). Do you think that would be helpful enough?

Reply:
Yeah, that would make sense ^^

Reply:
I added a small note about these two dependencies on our contribution guide https://github.com/grafana/agent/pull/1478/commits/323fb0084cadcf6a62a44c712a4fa8df5c2ef5ed.

Do we still want to have a `noebpf` flag to make it easier for people to contribute?